### PR TITLE
Add Sort feature into DG

### PR DESCRIPTION
**Pull request by angrybunny123**
Edited find feature and add feature. Updated add feature to reflect that sorting by user preference occurs after adding.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -257,14 +263,58 @@ Command : `find n/ Harry`
...
+##### Aspect: Sorting visible observable list or internal observable list
+
+* **Alternative 1 (current choice):** Sorting internal observable list
+  * Pros: Allows user to set sorting preference. When the user exits and reopens the application, the books would remain
+  in the sorted order.
+  * Cons: Could be a drawback if the user only wants to sort for temporary view.
+  
+* **Alternative 2:** Sorting visible observable list
+  * Pros: Sorts the user's book list temporarily, which is useful if the user only wants the sorted view momentarily.
+  * Cons: Book list would always return to default view after subsequent commands which is be distracting.
```
Comment (TanLeYang) [inline docs/DeveloperGuide.md]: Typo here? the "which is be distracting" part

**Review by TanLeYang — COMMENTED** [review]
LGTM except for the small typo!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (angrybunny123) [reply docs/DeveloperGuide.md]: Yup that's a typo, nice catch!

**Review by pennhanlee — APPROVED** [review]
LGTM

**Pull request merged**